tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/Sources.cs

```diff
@@ -35,6 +35,40 @@ void BuildInstrumentationDefinitions(StringBuilder sb, List<CallTargetDefinition
             }};");
             }
 
+            static void BuildInstrumentedAssemblies(StringBuilder sb, List<CallTargetDefinitionSource> orderedDefinitions)
+            {
+                var lastAssembly = string.Empty;
+                foreach (var orderedDefinition in orderedDefinitions)
+                {
+                    if (orderedDefinition.AssemblyName == lastAssembly || IsKnownAssemblyPrefix(orderedDefinition.AssemblyName))
+                    {
+                        continue;
+                    }
+
+                    lastAssembly = orderedDefinition.AssemblyName;
+                    sb.Append('"')
+                      .Append(lastAssembly)
+                      .Append("\", ");
+                }
+
+                // NOTE: Keep this in sync with ExceptionRedactor.IsKnownAssemblyPrefix()
+                static bool IsKnownAssemblyPrefix(string assemblyName)
+                {
+                    return assemblyName.StartsWith("Datadog.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("mscorlib,", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("Microsoft.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("System.", StringComparison.Ordinal)
```

Comment:
Should we exclude `System` as well? I see it twice in the generated code

Reply:
So, we _could_ add an extra `assemblyName.Equals("System")`, but it doesn't seem worth adding that special case when it ends up in the generated code twice anyway 🤷‍♂️ As for the duplicate `"System"` entry, thanks, that's a bug in my de-duplication code 🙂 I was trying to avoid doing too much extra Linq, but it's probably easier to just suck it up 😄

Reply:
Fixed so that we 
1. Specify the size of the HashSet upfront (as we know it)
2. Deduplicate and sort the keys we add